src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrayClone = [];
+  const stateClone = { ...state };
+
+  for (const action of actions) {
```

Comment:
you can use destructuring of objects:  `const { type, extraData, keysToRemove } = action;`

after switch case will be : 
'    switch (type) {
      case 'addProperties':
        Object.assign(stateClone, extraData);
        break;

      case 'removeProperties':
        for (const key of keysToRemove) {
          delete stateClone[key];
        }
        break;

      case 'clear':
        for (const key in stateClone) {
          delete stateClone[key];
        }
        break;

      default:
        break;
    }' 

or you can write this sorter (I known about this today at the lesson) '
  for (const { type, extraData, keysToRemove } of actions) {
    // const { type, extraData, keysToRemove } = action;'

Reply:
GOOD JOB!!!!